solr/core/src/java/org/apache/solr/core/ZkContainer.java

```diff
@@ -144,25 +139,6 @@ public void initZooKeeper(final CoreContainer cc, CloudConfig config) {
           } else {
             UrlScheme.INSTANCE.setUrlScheme(System.getProperty(URL_SCHEME, HTTP));
           }
-
-          if (zkServer.getServers().size() > 1 && confDir == null && boostrapConf == false) {
-            // we are part of an ensemble and we are not uploading the config - pause to give the config time
-            // to get up
-            Thread.sleep(10000);
```

Comment:
hi @markrmiller, I encountered this logic, while moving these `bootstrap_confdir ` and `bootstrap_conf` properties to ConfigSetService. It seems to me this logic is no longer valid and removed it.  Could you elaborate more on this logic if it is needed?

Reply:
Looks like very old behavior for embedded zookeeper mode when you had to preconfigure a collection before startup.

I don't know the state of embedded zk, but I have a hard time imagining this makes any sense anymore. If it did, it's a terrible way to wait.

I'm sure it did this because with a preconfigured collection, the cores coming up would need to find config or fail. Nothing works this way anymore. Default config is uploaded automatically. You start with 0 cores or collections. 

Embedded zk was never part of any test as far as I know, but id assume this bootstrap stuff is irrelevant now.